hack/main.go.tmpl

```diff
@@ -60,7 +67,11 @@ func main() {
 		essTLSCertsPath            = app.Flag("ess-tls-cert-dir", "Path of ESS TLS certificates.").Envar("ESS_TLS_CERTS_DIR").String()
 		enableManagementPolicies   = app.Flag("enable-management-policies", "Enable support for Management Policies.").Default("true").Envar("ENABLE_MANAGEMENT_POLICIES").Bool()
 
-		certsDir = app.Flag("certs-dir", "The directory that contains the server key and certificate.").Default("/tls/server").Envar("CERTS_DIR").String()
+		certsDirSet = false
+		certsDir    = app.Flag("certs-dir", "The directory that contains the server key and certificate.").Default(tlsServerCertDir).Envar(certsDirEnvVar).PreAction(func(_ *kingpin.ParseContext) error {
```

Comment:
nit: As discussed offline, here we rely on calling the PreAction mechanism when the flag is set. I am writing this to record it for documentation purposes for the other contributors. We may consider putting a code comment here.

Reply:
Thanks @sergenyalcin. Added comments that explain the purpose of using a `PreAction` there.